lib/CXGN/Dataset.pm

```diff
@@ -339,6 +338,76 @@ sub get_datasets_by_user {
     return \@datasets;
 }
 
+=head2 datasets_public()
+
+=cut
+
+sub get_datasets_public {
+    my $class = shift;
+    my $people_schema = shift;
+
+    my $rs = $people_schema->resultset("SpDataset")->search( { is_public => 1 });
+
+    my @datasets;
+    while (my $row = $rs->next()) {
+        push @datasets,  [ $row->sp_dataset_id(), $row->name(), $row->description() ];
+    }
+
+    return \@datasets;
+}
+
+=head2 datasets_public()
+
+=cut
+
+sub set_dataset_public {
+    my $self = shift;
+
+    my $row = $self->people_schema()->resultset("SpDataset")->find( { sp_dataset_id => $self->sp_dataset_id() });
+
+    if (! $row) {
+        return "The specified dataset does not exist";
+    } else {
+        eval {
+	   $row->is_public(1);
```

Comment:
Were there changes to the CXGN::People::Schema? If so where can one find those changes? Thanks!